scripts/docker.sh

```diff
@@ -54,7 +54,7 @@ function docker_build {
   docker build \
     -t "${GCR_REGISTRY}/${name}:${UNSTABLE_TAG}" \
     -f "${dockerfile}" \
-    "${build_args[@]}" \
+    "${build_args[@]+"${build_args[@]}"}" \
```

Comment:
Can you please explain what are we doing here?

Reply:
The syntax is a bit arcane, I agree :)

From https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html:
```
${parameter:+word}
If parameter is null or unset, nothing is substituted, otherwise the expansion of word is substituted.
```

So basically if `build_args` is unset or null, it will substitute nothing. Otherwise, it will expand the `build_args` array.

According to [this stackoverflow post](https://stackoverflow.com/questions/7577052/bash-empty-array-expansion-with-set-u) this is the safest way to achieve this across bash versions.

I can also add a comment to help future readers

Reply:
Thanks for you explanation